test/test_torch.py

```diff
@@ -14185,6 +14181,22 @@ def test_atanh_domain_float(self, device, dtype):
         self.assertEqual(torch.isinf(torch.atanh(sample)), inf_mask)
         self.assertEqual(torch.isinf(sample.atanh()), inf_mask)
 
+    def test_nullary_op_mem_overlap(self, device):
+        ops = (
+            ("random_", ()),
+            ("uniform_", ()),
+            ("cauchy_", ()),
+            ("log_normal_", ()),
+            ("exponential_", ()),
+            ("geometric_", (0.5,)),
+            ("normal_", ()),
```

Comment:
We should add a test for bernoulli_ too: https://github.com/pytorch/pytorch/blob/284ff047927e3b7f44f17a28f62c796bca93cea3/aten/src/ATen/native/cuda/DistributionBernoulli.cu#L38
https://github.com/pytorch/pytorch/blob/284ff047927e3b7f44f17a28f62c796bca93cea3/aten/src/ATen/native/cpu/DistributionTemplates.h#L345

Some versions of bernoulli go through different code paths :/ we should probably try to keep them consistent in a single PR: https://github.com/pytorch/pytorch/blob/284ff047927e3b7f44f17a28f62c796bca93cea3/aten/src/ATen/native/cpu/DistributionTemplates.h#L318-L322

Reply:
(see https://github.com/pytorch/pytorch/pull/43421#issuecomment-679386089)